sdk/opendp/smartnoise/_ast/tokens.py

```diff
@@ -129,6 +129,58 @@ def find_nodes(self, type_name, not_child_of=None):
         childnodes = [c.find_nodes(type_name, not_child_of) for c in sqlnodes]
         return nodes + flatten(childnodes)
 
+    def visualize(self, color_types={}, n_trunc=None):
+        """
+        Construct the AST graph of the object
+
+        Args:
+            color_type (optional): A dictionnary which contains the type and
+                the color of the nodes.
+                If no color is provided, the displayed node will be black.
+                Example: {str: 'red', Query: 'green'}
+            n_trunc: for visibility, truncate the expressions.
+                By default, there is no truncation.
+
+        Returns:
+            graphviz Digraph
+
+        """
+        def _label_node(expr, n_trunc):
+            str_expr = str(expr)
+            if n_trunc and len(str_expr) > n_trunc:
+                str_expr = str_expr[:n_trunc] + '...'
+            return f"{type(expr).__name__}: {str_expr}"
```

Comment:
I think this syntax is only supported in Python 3.6+, but it should be OK, since our SQLite dependency for Pandas support now requires 3.6+.